tools/mksurfdata_map/src/mkpftUtilsMod.F90

```diff
@@ -251,6 +252,39 @@ subroutine adjust_total_veg_area(new_total_pct, pctnatpft, pctcft)
 
   end subroutine adjust_total_veg_area
 
+  !-----------------------------------------------------------------------
+  subroutine adjust_total_natveg_area(new_total_pct, pctnatpft)
```

Comment:
Done.  I replaced the subroutine with this call in the main subroutine (mksurfdata):

       call pctnatpft(n)%set_pct_l2g(new_total_natveg_pct)

This results in some roundoff changes in surface dataset fields, e.g., with and without this change (e.g., a 1deg 1850 surface dataset):

 RMS PCT_NATVEG                       1.3323E-15            NORMALIZED  4.8053E-17
 RMS PCT_CROP                         1.0764E-16            NORMALIZED  1.0965E-16
 RMS PCT_NAT_PFT                      1.5614E-17            NORMALIZED  2.3421E-18
 RMS PCT_LAKE                         1.1245E-16            NORMALIZED  1.9502E-16
 RMS PCT_GLACIER                      5.6078E-17            NORMALIZED  4.4766E-18
 RMS PCT_URBAN                        2.5112E-17            NORMALIZED  4.1348E-16

Presumably this is because the argument in this line (a multiply, divide):

call pctnatpft%set_pct_l2g(natpft_l2g * new_total_pct / old_total)
(Note that the subroutine is called with an argument of new_total_natveg_pct which is assigned to new_total_pct)

is not bfb the same as:

call pctnatpft(n)%set_pct_l2g(new_total_natveg_pct).

Reply:
Great, thanks. I'm not surprised by the roundoff-level changes; I agree with your explanation. Thanks for your careful testing & documentation of the change.